### PR TITLE
Implemented getColumnNames getter 

### LINKED ISSUE #255: create getter to get existing column names
- [x] when called, returns an array of strings of all column names in the provided data table (or in the data dictionary if equivalent)

Other requirements:

- [x] depending components have been checked and are calling this getter correctly
- [x] have unit tests to check functionality
- [x] needed store state variables exist already or have been created if not

### PR DESCRIPTION
**Pull request by rmanaem**
Closes #255

### CODE REVIEW

File: cypress/component/column-linking-table.cy.js [hunk 1]
```diff
@@ -98,7 +97,8 @@ describe("The column-linking-table component", () => {
     it("can alter link relation (add/remove) between a column and a category", () => {
 
         // 0. The first category and column
-        const participantIDColumn = store.getters.columns()[0].name;
+        const participantIDColumn = store.getters.getColumnNames()[0];
+        console.log(participantIDColumn);
```
Comment (surchs) [inline cypress/component/column-linking-table.cy.js:101]: leftover debug?

File: cypress/unit/store-getter-getColumnNames.cy.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { getters } from "~/store/index-refactor";
+
+describe("getColumnNames", () => {
+    it("Returns an array of all column names in the data table", () => {
+        const { getColumnNames } = getters;
+    const state = {
```
Comment (surchs) [inline cypress/unit/store-getter-getColumnNames.cy.js:6]: indentation looks odd to me

File: package.json [hunk 3]
```diff
@@ -8,7 +8,7 @@
     "start": "nuxt start",
     "generate": "nuxt generate",
     "lint": "npx eslint --ext .vue  --ext .js .",
-    "prepare": "husky install",
+    "lint-fix": "npx eslint --ext .vue  --ext .js . --fix",
```
Comment (surchs) [inline package.json:11]: was this done by accident? the `lint-fix` script looks cool, but that should done in a separate PR. And we probably don't want to remove the `prepare` script unless we know it's not needed.

File: store/index-refactor.js [hunk 4]
```diff
@@ -35,7 +36,12 @@ export const getters = {
             return "";
         }
         return description;
+    },
+
+    getColumnNames(p_state) {
+        return (0 === p_state.dataTable.length) ? [] : Object.keys(p_state.dataTable[0]);
```
Comment (surchs) [inline store/index-refactor.js:42]: I mentioned the same thing in #291: is there any reason to expect that this `getter` is called before `dataTable` is set to something non-empty? If yes, we should decide whether that's something that should work or not break. And then we should test that our components are still happy if they get an empty array back. 

My sense is that we don't need to worry about this unless there is a good reason this getter could be called without `dataTable` being set. But let me know if you disagree.

**Review by surchs — CHANGES_REQUESTED** [review]
Thanks for the PR @rmanaem. This all looks very good. However #291 has implemented essentially the same getter (minus the tests and component fixes). So I think we should have a chat with @jarmoza on how to proceed with this. 

My sense is that we can probably merge #291 without implementing this getter, but for the moment I would wait for us to chat :).

File: cypress/component/column-linking-table.cy.js (hunk 1 above)
Comment (rmanaem) [reply cypress/component/column-linking-table.cy.js:101]: Yea good catch

**Comment by rmanaem** [comment]
> Thanks for the PR @rmanaem. This all looks very good. However #291 has implemented essentially the same getter (minus the tests and component fixes). So I think we should have a chat with @jarmoza on how to proceed with this.

Yea, I saw Jonathan's PR and comment on #255 when working on this so I replaced my implementation with his to avoid unnecessary conflicts. If I recall correctly he also wrote a test for the getColumnNames getter.

> My sense is that we can probably merge #291 without implementing this getter, but for the moment I would wait for us to chat :).

I also think we should remove(or move) the getColumnNames test from the createColumnToCategoryMap unit test, but let's discuss both of these issues with Jonathan.

**Comment by jarmoza** [comment]
@rmanaem I agree and will remove the `getColumnNames` test from my PR.

**Review by surchs — APPROVED** [review]
:tada: good to go! :eyes:

File: store/index-refactor.js (hunk 4 above)
Comment (rmanaem) [reply store/index-refactor.js:54]: Answered by Jonathan in #291
